src/nexus.cpp

```diff
@@ -430,3 +432,26 @@ void Nexus::bringAllToFront()
     focused->raise();
 }
 #endif
+
+void Nexus::screenshot()
+{
+    if (widget == nullptr) {
+        qWarning() << "No widget to take screenshot of";
+        return;
+    }
+
+    // Paint the widget onto a pixmap. This makes sure the screenshot still
+    // works even if we don't have screen capture permissions.
+    QPixmap pixmap(widget->size());
+    widget->render(&pixmap);
+
+    QString path = QStandardPaths::writableLocation(QStandardPaths::PicturesLocation);
+    if (path.isEmpty()) {
+        path = QDir::currentPath();
+    }
+    path += "/qtox-screenshot.png";
+    qDebug() << "Saving screenshot to" << path;
+    if (!pixmap.save(path)) {
+        qWarning() << "Failed to save screenshot";
+    }
+}
```

Comment:
can we make this a no-op if a specific env var is not set? this kind of feels like it could be abused.

Reply:
Done.